pkg/agent/agent.go

```diff
@@ -642,13 +642,38 @@ func (i *Initializer) initNodeLocalConfig() error {
 		return err
 	}
 
-	ipAddr, err := k8s.GetNodeAddr(node)
-	if err != nil {
-		return fmt.Errorf("failed to obtain local IP address from K8s: %w", err)
-	}
-	localAddr, localIntf, err := getIPNetDeviceFromIP(ipAddr)
-	if err != nil {
-		return fmt.Errorf("failed to get local IPNet device with IP %v: %v", ipAddr, err)
+	var localAddr *net.IPNet
+	var localIntf *net.Interface
+	if i.networkConfig.InBandIface != "" {
+		// Find the configured in-band interface, and update its IP address in Node's annotation.
+		localAddr, localIntf, err = util.GetInterfaceByName(i.networkConfig.InBandIface, i.ovsBridge)
```

Comment:
I am not very certain it is good to reuse NodeConfig.NodeIP for the in-band address. Do we use NodeConfig.NodeIP for other purposes besides cross-Node traffic? Could you check?

Reply:
Having a deep think, I agree we use a seperate config for in-band interface is better. I would change.

Reply:
Ok. Consider TransportIP if we use TransportInterface for the interface name.